### DISCUSSION TITLE
Issue with paginationPageSizeSelector Implementation in AG Grid

### DISCUSSION THREAD

**Discussion by Swasthik-Jain** [zauberzeug/nicegui#3059] [Q&A]
### Question

I'm currently working on implementing pagination in AG Grid, and I'm encountering difficulties with the paginationPageSizeSelector option. Despite my efforts, I haven't been able to get it working as expected.
```py
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'sortable': True, 'align': 'left', 'text': 'bold',
     'filter': 'agTextColumnFilter', 'floatingFilter': True},
    {'class': 'class', 'label': 'Class', 'field': 'class', 'sortable': True, 'align': 'left',
     'filter': 'agTextColumnFilter', 'floatingFilter': True},
    {'iri': 'iri', 'label': 'IRI', 'field': 'iri', 'sortable': True, 'align': 'left',
     'filter': 'agTextColumnFilter', 'floatingFilter': True}]

rows = []
for result in response['_All'] or []:
    row = {
        'name': result['name'],
        'class': result['className'],
        'iri': result['externalID']
    }
    rows.append(row)

grid_data = {
    'columnDefs': columns,
    'rowData': rows,
    'pagination': True,
    'paginationPageSize': 20,
    'paginationPageSizeSelector': [10, 20, 50, 100]
}

ui.aggrid(grid_data).classes('grid-class')
```
        
I've set up the paginationPageSizeSelector to allow users to select the number of rows per page, but it doesn't seem to be functioning as intended.

I'm seeking guidance on how to correctly implement the paginationPageSizeSelector in AG Grid. Any insights, corrections, or alternative approaches to achieve this functionality would be greatly appreciated.

**Comment by falkoschindler** [comment]
Hi @Swasthik-Jain,

I reproduced the problem with the following code:
```py
ui.aggrid({
    'columnDefs': [{'name': 'name', 'label': 'Name', 'field': 'name'}],
    'rowData': [{'name': f'name{i}'} for i in range(100)],
    'pagination': True,
    'paginationPageSize': 20,
    'paginationPageSizeSelector': [10, 20, 50, 100],
})
```
Indeed, the page size selector isn't displayed. And there is a JavaScript warning about an unrecognized option "paginationPageSizeSelector".

It turns out this selector has only been added in AG Grid version 31.x.x, but NiceGUI still uses 30.2.0. We'll upgrade in NiceGUI 2.0, which is the next major milestone.

**Reply by Swasthik-Jain** [reply]
Thank you for letting me know about the issue .
